### ISSUE TITLE
Multi classification task 

### ISSUE THREAD

**Issue by heniar** [materialsvirtuallab/megnet#322]
Hello, 

I meet a reproducible issue when doing a multi-classification task. I have a dataset of 25k items, which include five classes of 17k, 2k, 2k, 2k, 2k  items. With the same dataset, the random forest can provide a cross-validation test accuracy of 0.832786252172234. 

I'm using megnet 1.2.8 from PyPI, during training, the loss became negative in  Epoch 1/500 ( 1/162 ETA: 0s - loss: 0.6950; 8/162 ETA: 4:02 - loss: 0.1071; 9/162  ETA: 4:05 - loss: -0.0201) and keep decrsing in the following epoch. Finaly a model with very high negative loss was produced, the prediction "model.predict_graph(graph)" was always "array([1.], dtype=float32)" . 

Here are my codes:
```
from megnet.models import MEGNetModel
from megnet.data.crystal import CrystalGraph
import numpy as np

nfeat_bond = 100
epochs = 500

r_cutoff = 4
gaussian_centers = np.linspace(0, r_cutoff + 1, nfeat_bond)
gaussian_width = 0.5
graph_converter = CrystalGraph(cutoff=r_cutoff)
model = MEGNetModel(graph_converter=graph_converter, centers=gaussian_centers, width=gaussian_width,
                    is_classification=True)

graphs_valid = []
targets_valid = []
structures_invalid = []
for s, p in zip(train['structure'], train[my_property]):
    try:
        graph = model.graph_converter.convert(s)
        graphs_valid.append(graph)
        targets_valid.append(p)
    except:
        structures_invalid.append(s)

model.train_from_graphs(graphs_valid, targets_valid, epochs=epochs)
```

Do you have any idea about what might cause this issue? 
Many thanks!

**Comment by chc273** [comment]
@heniar multiclass classification is not supported out of box. You will need to change the last layer to softmax, and change the loss function too.

**Comment by heniar** [comment]
I hacked the code to change https://github.com/materialsvirtuallab/megnet/blob/0c5ce2f97e3dce525a1685d96d52652fca20de10/megnet/models/megnet.py#L354
to be "softmax" 
and change 
https://github.com/materialsvirtuallab/megnet/blob/0c5ce2f97e3dce525a1685d96d52652fca20de10/megnet/models/megnet.py#L123
to be "categorical_crossentropy".

This time, the loss started from 3.8743e-07.

I also tested keeping loss function as "binary_crossentropy",  the loss started from -34.0725.

**Comment by heniar** [comment]
By the way, my structures are crystals, and  I use an int (0:4) as the target, should I do One-Hot Encoding to converse the target before inputting to the model?  (In my previous binary task, I use an int (0:1) as the target to train MEGNet could get good results)

**Comment by chc273** [comment]
Yes, use one-hot encoding and also change the number of prediction targets to 4 should work.

**Comment by heniar** [comment]
Thanks a lot, MEGNetModel(ntarget=5, ...) with one-hot encoding got good results
